src/main/java/de/zalando/zmon/dataservice/data/RedisDataStore.java

```diff
@@ -111,6 +115,11 @@ public void store(WorkerResult wr) {
                                 "else " +
                                     "return redis.call('sadd','zmon:alerts'," + alert.alert_id + ") " +
                                 "end");
+
+                        if (alert.alert_evaluation_ts != null) {
```

Comment:
However, it's a little bit inaccurate, because `duration` is calculated before the record is actually written to Redis (` p.sync()` below ensures that all the data is written). But I believe it's not that dangerous